src/hooks/airdrop/stateHelpers/useGetAirdropUserDetails.ts

```diff
@@ -71,13 +72,22 @@ export const useGetAirdropUserDetails = () => {
         });
     }, [handleRequest, setReferralCount]);
 
+    const fetchAcceptedReferral = useCallback(async () => {
+        const data = await handleRequest<{ claimed: boolean }>({
+            path: '/miner/claimed-referral',
+            method: 'GET',
+        });
+        setAcceptedReferral(!!data?.claimed);
+    }, [handleRequest, setAcceptedReferral]);
+
     useEffect(() => {
         const fetchData = async () => {
             const details = await fetchUserDetails();
             if (!details?.rank.gems) {
                 await fetchUserPoints();
             }
             await fetchUserReferralPoints();
+            await fetchAcceptedReferral();
```

Comment:
Should we use a promise.all to batch all these requests? If they're local only it's fine not to, if each of these are making web requests it might be worth batching it into a Promise.all